### DISCUSSION TITLE
Implement edit with modal

### DISCUSSION THREAD

**Discussion by mihalikv** [django-commons/django-unicorn#426] [Q&A]
Hi, 

I am playing around with django-unicorn and I am not sure how to integrate it properly with django ModelForm. Let's say that I want a list of objects with edit button and after a click on the edit button, modal with prefilled form is opened.

I started with a template like this:

```
{% load unicorn %}

<div>
  <div id="modal-1" class="modal fade" tabindex="-1" aria-labelledby="modal-1-btn" aria-hidden="true">
    <div class="modal-dialog">
      <div class="modal-content">
        <div class="modal-header">
          <h2 class="text-h3">title</h2>
        </div>
        <div class="modal-body">
          <div>
            {% if form %}
              {{ form.as_p }}
            {% endif %}
          </div>
        </div>
        <div class="modal-footer">
          <button unicorn:click="$validate" class="btn">Validate</button>
        </div>
      </div>
    </div>
  </div>
  <ul>
    {% for purchase in purchases %}
      <li>{{ purchase.purchase_number }}
        <button unicorn:click="set_form('{{ purchase.id }}')" id="modal-1-btn" type="button" class="btn" data-bs-toggle="modal" data-bs-target="#modal-1">
          Edit order
        </button>
      </li>
    {% endfor %}
  </ul>
</div>
```

and following view:

```
from django import forms
from django.shortcuts import get_object_or_404
from django_unicorn.components import UnicornView, QuerySetType

from sbcore.purchase.models import Purchase


class PurchaseForm(forms.ModelForm):
    class Meta:
        model = Purchase
        fields = ['purchase_number']


class PurchaseListView(UnicornView):
    purchases: QuerySetType[Purchase] = None
    form_class = PurchaseForm
    form = None

    def mount(self):
        self.purchases = Purchase.objects.all()

    def set_form(self, purchase_id):
        purchase = get_object_or_404(Purchase, pk=purchase_id)
        self.form = self.form_class(purchase)
```

I am aware of that form.as_p will not work and somehow I need to add `unicorn:model` to the input fields but first of all, I am not sure how to pre-fill the form with some data.

**Comment by adamghill** [comment]
Sorry, I missed this message from a while ago. There isn't a current way to use the built-in Django form-rendering, so you have to write the HTML manually unfortunately. https://docs.djangoproject.com/en/4.1/topics/forms/#rendering-fields-manually has information around that.

Instead of 
```
{{ form.as_p }}
```
you could do something like:
```
<input unicorn:model="form.purchase_number" type="text"></input>
```
That should take the `form` in `UnicornView` and bind the input to `form.purchase_number`. Let me know if that doesn't work or doesn't make sense -- I can add a sample to my examples if it would be useful.

**Comment by LexxLuey** [comment]
Hello there.
I tried out the code above because I needed to use a modal to edit data just like @mihalikv but I get an error. Here is my code:

`#memberslist.py
from django_unicorn.components import UnicornView, QuerySetType
from django import forms
from django.shortcuts import get_object_or_404

from pprint import pprint

from members.models import Kid
from core.constants import AGE_GROUP_CLASSES, SEXES, STATES


class EditForm(forms.ModelForm):
    class Meta:
        model = Kid
        fields = ("__all__")

class MemberslistView(UnicornView):
    kids: QuerySetType[Kid] = Kid.objects.none()
    first_name: str = ""
    last_name: str = ""
    middle_name: str = ""
    nick_name: str = ""
    address: str = ""
    join_date: str = "10/05/2018"
    dob: str = "10/05/2018"
    age_group_classes: list = []
    sexes: list = []
    states: list = []
    selected_age_group_class: int = 4
    selected_sex: int = 1
    selected_state: int = 37

    form_class = EditForm
    form = None

    class Meta:
        javascript_exclude = (
            "age_group_classes",
            "sexes",
            "states",
        )


    def mount(self):
        """On mount, populate the kids property w/ a QuerySet of all kids"""
        self.kids = Kid.objects.all()
        self.age_group_classes = AGE_GROUP_CLASSES
        self.sexes = SEXES
        self.states = STATES
        print(self.kids)

    def add_kid(self):
        """Create the new Kid, get new list of all kids, and clear the 'name' property"""
        print()
        Kid.objects.create(
            first_name=self.first_name,
            last_name=self.last_name,
            middle_name=self.middle_name,
            nick_name=self.nick_name,
            address=self.address,
            join_date=self.join_date,
            dob=self.dob,
            age_group_class=self.selected_age_group_class,
            sex=self.selected_sex,
            state_of_origin=self.selected_state,
        )
        self.kids = Kid.objects.all()
        self.first_name = ""
        self.last_name = ""
        self.middle_name = ""
        self.nick_name = ""
        self.address = ""
        self.join_date: str = "10/05/2018"
        self.dob: str = "10/05/2018"
        self.age_group_classes: list = AGE_GROUP_CLASSES
        self.sexes: list = SEXES
        self.states: list = STATES
        self.selected_age_group_class: int = 5
        self.selected_sex: int = 3
        self.selected_state: int = 37

    def delete_all(self):
        """Delete all kids and reset 'kids' property"""
        Kid.objects.all().delete()
        self.kids = Kid.objects.none()

    def set_form(self, id):
        kid = get_object_or_404(Kid, pk=id)
        self.form = self.form_class(kid)`

`#models.py
from django.db import models
from core.mixins import ModelMixin


class Kid(ModelMixin):
    class Sex(models.IntegerChoices):
        MALE = 1
        FEMALE = 2
        UNKNOWN = 3

    class AgeGroupClasses(models.IntegerChoices):
        CLIMBERS = 1
        TRAILBLAZERS = 2
        CHAMPIONS = 3
        WORLDCHANGERS = 4
        UNKNOWN = 5
    
    class State(models.IntegerChoices):
        Abia = 1
        Adamawa = 2
        Akwa_Ibom = 3
        Anambra = 4
        Bauchi = 5
        Bayelsa = 6
        Benue = 7
        Borno = 8
        Cross_River = 9
        Delta = 10
        Ebonyi = 11
        Edo = 12
        Ekiti = 13
        Enugu = 14
        Gombe = 15
        Imo = 16
        Jigawa = 17
        Kaduna = 18
        Kano = 19
        Katsina = 20
        Kebbi = 21
        Kogi = 22
        Kwara = 23
        Lagos = 24
        Nasarawa = 25
        Niger = 26
        Ogun = 27
        Ondo = 28
        Osun = 29
        Oyo = 30
        Plateau = 31
        Rivers = 32
        Sokoto = 33
        Taraba = 34
        Yobe = 35
        Zamfara = 36
        FCT = 37
        Unknown = 38

    
    first_name = models.CharField(max_length=200, null=True, blank=True)
    last_name = models.CharField(max_length=200, null=True, blank=True)
    middle_name = models.CharField(max_length=200, null=True, blank=True)
    nick_name = models.CharField(max_length=200, null=True, blank=True)
    sex = models.IntegerField(
        choices=Sex.choices, default=Sex.UNKNOWN
    )
    age_group_class = models.IntegerField(
        choices=AgeGroupClasses.choices, default=AgeGroupClasses.UNKNOWN
    )
    dob = models.DateField(null=True, blank=True)
    join_date = models.DateField(null=True, blank=True)
    state_of_origin = models.IntegerField(
        choices=State.choices, default=State.Unknown
    )
    address = models.CharField(max_length=500, null=True, blank=True)

    context_object_name = 'kids'
    class Meta:
        verbose_name = "kid"
        verbose_name_plural = "kids"
`

` 
<div>
    
    <div class="row">
        <div class="modal fade bd-example-modal-lg" tabindex="-1" role="dialog" aria-labelledby="myLargeModalLabel" aria-hidden="true" id="editModal">
            <div class="modal-dialog modal-lg">
                <div class="modal-content">
                    <div class="modal-content">
                        <div class="modal-header">
                            <h5 class="modal-title" id="exampleModalLabel">Edit Data</h5>
                            <button type="button" class="close" data-dismiss="modal" aria-hidden="true">
                                <i class="tim-icons icon-simple-remove"></i>
                            </button>
                        </div>
                        <div class="modal-body">
                            <div class="card">
                                <div class="card-body">
                                    <div class="form-row">
                                        <div class="form-group col-md-4">
                                            <label for=""><b> FIRST NAME</b></label>
                                            <div class="input-group">
                                                <div class="input-group-prepend">
                                                    <div class="input-group-text">
                                                        <i class="tim-icons icon-single-02"></i>
                                                    </div>      
                                                </div>
                                                <input unicorn:model="form.first_name" class="form-control" />
                                            </div>
                                        </div>
                                        <div class="form-group col-md-4">
                                            <label for=""><b> LAST NAME</b></label>
                                            <div class="input-group">
                                                <div class="input-group-prepend">
                                                    <div class="input-group-text">
                                                        <i class="tim-icons icon-single-02"></i>
                                                    </div>      
                                                </div>
                                                {% comment %} <input type="text" unicorn:model.defer="last_name" class="form-control" placeholder="Enter Last Name" /> {% endcomment %}
                                            </div>
                                        </div>
                                        <div class="form-group col-md-4">
                                            <label for=""><b> MIDDLE NAME</b></label>
                                            <div class="input-group">
                                                <div class="input-group-prepend">
                                                    <div class="input-group-text">
                                                        <i class="tim-icons icon-single-02"></i>
                                                    </div>      
                                                </div>
                                                {% comment %} <input type="text" unicorn:model.defer="middle_name" class="form-control" placeholder="Enter Middle Name" /> {% endcomment %}
                                            </div>
                                        </div>
                                    </div>
                                    <div class="form-row">
                                        <div class="form-group col-md-4">
                                            <label for=""><b> NICK NAME</b></label>
                                            <div class="input-group">
                                                <div class="input-group-prepend">
                                                    <div class="input-group-text">
                                                        <i class="tim-icons icon-badge"></i>
                                                    </div>      
                                                </div>
                                                {% comment %} <input type="text" unicorn:model.defer="nick_name" class="form-control" placeholder="Enter Nick Name" /> {% endcomment %}
                                            </div>
                                        </div>
                                        <div class="form-group col-md-4">
                                            <label for="inputState"><b>STATE OF ORIGIN</b></label>
                                            <select unicorn:model.defer="selected_state" id="inputState" class="form-control">
                                                <option selected>Choose...</option>
                                                {% for state in states %}
                                                <option value={{state.value}} >{{state.name}}</option>
                                                {% endfor %}
                                            </select>
                                        </div>
                                        <div class="form-group col-md-4">
                                            <label for=""><b> ADDRESS</b></label>
                                            <div class="input-group">
                                                <div class="input-group-prepend">
                                                    <div class="input-group-text">
                                                        <i class="tim-icons icon-square-pin"></i>
                                                    </div>      
                                                </div>
                                                {% comment %} <input type="text" unicorn:model.defer="address" class="form-control" placeholder="Enter Address" /> {% endcomment %}
                                            </div>
                                        </div>
                                    </div>
                                    <div class="form-row">
                                        <div class="form-group col-md-3">
                                            <label for=""><b> SEX</b></label>
                                            <select id="inputSex" unicorn:model.defer="selected_sex" class="form-control">
                                                <option selected>Choose...</option>
                                                {% for sex in sexes %}
                                                <option value={{sex.value}} >{{sex.name}}</option>
                                                {% endfor %}
                                            </select>
                                        </div>
                                        <div class="form-group col-md-3">
                                            <label for=""><b> {{'Class'|upper}} </b></label>
                                            <select unicorn:model.defer="selected_age_group_class" id="inputClass" class="form-control">
                                                <option selected>Choose...</option>
                                                {% for age_group_class in age_group_classes %}
                                                <option value={{age_group_class.value}} >{{age_group_class.name}}</option>
                                                {% endfor %}
                                            </select>
                                        </div>
                                        <div class="form-group col-md-3">
                                            <label for=""><b> {{'Date Of Birth'|upper}}</b></label>
                                            <div class="input-group">
                                                <div class="input-group-prepend">
                                                    <div class="input-group-text">
                                                        <i class="tim-icons icon-calendar-60"></i>
                                                    </div>      
                                                </div>
                                                {% comment %} <input type="date" class="form-control" unicorn:model.defer="dob"/> {% endcomment %}
                                            </div>
                                        </div>
                                        <div class="form-group col-md-3">
                                            <label for=""><b> {{'Join Date'|upper}}</b></label>
                                            <div class="input-group">
                                                <div class="input-group-prepend">
                                                    <div class="input-group-text">
                                                        <i class="tim-icons icon-calendar-60"></i>
                                                    </div>      
                                                </div>
                                                {% comment %} <input type="date" unicorn:model.defer="join_date" class="form-control" /> {% endcomment %}
                                            </div>
                                        </div>
                                    </div>
                                    {% comment %} <div class="form-group">
                                        <div class="form-check">
                                            <label class="form-check-label">
                                                <input class="form-check-input" type="checkbox" value="">
                                                Check me out
                                                <span class="form-check-sign">
                                                    <span class="check"></span>
                                                </span>
                                            </label>
                                        </div>
                                    </div> {% endcomment %}
                                </div>
                            </div>
                        </div>
                        <div class="modal-footer">
                            <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
                            <button type="button" class="btn btn-primary" unicorn:click="edit_kid">Save changes</button>
                        </div>
                    </div>
                </div>
            </div>
        </div>

        {% comment %} Form for creation of new kid {% endcomment %}
        <div id="accordion" role="tablist" aria-multiselectable="true" class="card-collapse">
            <div class="card card-plain">
                <div class="card-header" role="tab" id="headingTwo">
                    <a class="btn btn-primary" data-toggle="collapse" href="#addChild" aria-expanded="false" aria-controls="addChild">                        
                        <b>Add A New Child </b>                        
                    </a>
                </div>
                <div class="collapse" id="addChild">                    
                    <div class="col-md-12">
                        <div class="card ">
                            <div class="card-header">
                            </div>
                            <div class="card-body">
                            <div class="table-responsive">
                                <table class="table tablesorter " id="">
                                <thead class=" text-primary">
                                    <tr>
                                    <th>
                                    </th>
                                    <th>
                                    </th>
                                    <th>
                                    </th>
                                    <th class="text-center">
                                    </th>
                                    <th class="text-center">
                                        
                                    </th>
                                    </tr>
                                </thead>
                                <tbody>
                                    <tr>
                                    <td>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-single-02"></i>
                                                </div>      
                                            </div>
                                            <input type="text" disabled  class="form-control" placeholder="With Font Awesome Icons" />
                                        </div>
                                        {% comment %} <input type="text" unicorn:model="first_name" /> {% endcomment %}
                                    </td>
                                    <td>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-single-02"></i>
                                                </div>      
                                            </div>
                                            <input type="text" disabled  class="form-control" placeholder="With Font Awesome Icons" />
                                        </div>
            
                                        {% comment %} <input type="text" unicorn:model="last_name" /> {% endcomment %}
                                    </td>
                                    <td>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-single-02"></i>
                                                </div>      
                                            </div>
                                            <input type="text" disabled  class="form-control" placeholder="With Font Awesome Icons" />
                                        </div>
            
                                        {% comment %} <input type="text" unicorn:model="middle_name" /> {% endcomment %}
                                    </td>
                                    <td class="text-center">
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-single-02"></i>
                                                </div>      
                                            </div>
                                            <input type="text" disabled  class="form-control" placeholder="With Font Awesome Icons" />
                                        </div>
            
                                        {% comment %} <input type="text" unicorn:model="address" /> {% endcomment %}
                                    </td>
                                    <td class="text-center">
                                        <div class="input-group">
                                        </div>
                                        {% comment %} <button unicorn:click="add_kid">Add Movie</button> {% endcomment %}
            
                                        
                                        {% comment %} <button unicorn:click="delete_all">Delete All</button> {% endcomment %}
                                    </td>
                                    </tr>
                                </tbody>
                                </table>
                            </div>
                            </div>
                        </div>
                        
                        <div class="card">
                            <div class="card-body">
                                <div class="form-row">
                                    <div class="form-group col-md-4">
                                        <label for=""><b> FIRST NAME</b></label>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-single-02"></i>
                                                </div>      
                                            </div>
                                            <input type="text" unicorn:model.defer="first_name" class="form-control" placeholder="Enter First Name" />
                                        </div>
                                    </div>
                                    <div class="form-group col-md-4">
                                        <label for=""><b> LAST NAME</b></label>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-single-02"></i>
                                                </div>      
                                            </div>
                                            <input type="text" unicorn:model.defer="last_name" class="form-control" placeholder="Enter Last Name" />
                                        </div>
                                    </div>
                                    <div class="form-group col-md-4">
                                        <label for=""><b> MIDDLE NAME</b></label>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-single-02"></i>
                                                </div>      
                                            </div>
                                            <input type="text" unicorn:model.defer="middle_name" class="form-control" placeholder="Enter Middle Name" />
                                        </div>
                                    </div>
                                </div>
                                <div class="form-row">
                                    <div class="form-group col-md-4">
                                        <label for=""><b> NICK NAME</b></label>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-badge"></i>
                                                </div>      
                                            </div>
                                            <input type="text" unicorn:model.defer="nick_name" class="form-control" placeholder="Enter Nick Name" />
                                        </div>
                                    </div>
                                    <div class="form-group col-md-4">
                                        <label for="inputState"><b>STATE OF ORIGIN</b></label>
                                        <select unicorn:model.defer="selected_state" id="inputState" class="form-control">
                                            <option selected>Choose...</option>
                                            {% for state in states %}
                                            <option value={{state.value}} >{{state.name}}</option>
                                            {% endfor %}
                                        </select>
                                    </div>
                                    <div class="form-group col-md-4">
                                        <label for=""><b> ADDRESS</b></label>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-square-pin"></i>
                                                </div>      
                                            </div>
                                            <input type="text" unicorn:model.defer="address" class="form-control" placeholder="Enter Address" />
                                        </div>
                                    </div>
                                </div>
                                <div class="form-row">
                                    <div class="form-group col-md-3">
                                        <label for=""><b> SEX</b></label>
                                        <select id="inputSex" unicorn:model.defer="selected_sex" class="form-control">
                                            <option selected>Choose...</option>
                                            {% for sex in sexes %}
                                            <option value={{sex.value}} >{{sex.name}}</option>
                                            {% endfor %}
                                        </select>
                                    </div>
                                    <div class="form-group col-md-3">
                                        <label for=""><b> {{'Class'|upper}} </b></label>
                                        <select unicorn:model.defer="selected_age_group_class" id="inputClass" class="form-control">
                                            <option selected>Choose...</option>
                                            {% for age_group_class in age_group_classes %}
                                            <option value={{age_group_class.value}} >{{age_group_class.name}}</option>
                                            {% endfor %}
                                        </select>
                                    </div>
                                    <div class="form-group col-md-3">
                                        <label for=""><b> {{'Date Of Birth'|upper}}</b></label>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-calendar-60"></i>
                                                </div>      
                                            </div>
                                            <input type="date" class="form-control" unicorn:model.defer="dob"/>
                                        </div>
                                    </div>
                                    <div class="form-group col-md-3">
                                        <label for=""><b> {{'Join Date'|upper}}</b></label>
                                        <div class="input-group">
                                            <div class="input-group-prepend">
                                                <div class="input-group-text">
                                                    <i class="tim-icons icon-calendar-60"></i>
                                                </div>      
                                            </div>
                                            <input type="date" unicorn:model.defer="join_date" class="form-control" />
                                        </div>
                                    </div>
                                </div>
                                {% comment %} <div class="form-group">
                                    <div class="form-check">
                                        <label class="form-check-label">
                                            <input class="form-check-input" type="checkbox" value="">
                                            Check me out
                                            <span class="form-check-sign">
                                                <span class="check"></span>
                                            </span>
                                        </label>
                                    </div>
                                </div> {% endcomment %}
                                <button class="btn btn-primary" unicorn:click="add_kid">Create</button>
                            </div>
                        </div>
                    </div>
                    
                </div>
                
            </div>
        </div>

        {% comment %} List of Kids {% endcomment %}
        <div class="col-md-12">
            <div class="card ">
                <div class="card-header">
                <h3 class="card-title"><b> CFCIC KIDS </b></h3>
                </div>
                <div class="card-body">
                <div class="table-responsive">
                    <table class="table tablesorter " id="">
                    <thead class=" text-primary">
                        <tr>
                        <th>
                            Name
                        </th>
                        <th>
                            Nick Name
                        </th>
                        <th>
                            Address
                        </th>
                        <th class="text-center">
                            Class
                        </th>
                        <th class="text-center">
                            Actions
                        </th>
                        </tr>
                    </thead>
                    <tbody>
                        {% for kid in kids %}
                        <tr>
                            <td>
                                {{kid.first_name|title}} {{kid.middle_name|title}} {{kid.last_name|title}}
                            </td>
                            <td>
                                {{kid.nick_name|title}}
                            </td>
                            <td>
                                {{kid.address|title}}
                            </td>
                            <td class="text-center">
                                {% if kid.age_group_class == 1 %}
                                {{"climbers"|title}}
                                {% elif kid.age_group_class == 2 %}
                                {{"trailblazers"|title}}
                                {% elif kid.age_group_class == 3 %}
                                {{"champions"|title}}
                                {% elif kid.age_group_class == 4 %}
                                {{"world changers"|title}}
                                {% else %}
                                {{"unknown"|title}}
                                {% endif %}
                            </td>
                            <td class="text-center">
                                <button type="button" class="btn btn-primary btn-fab btn-icon btn-round" data-toggle="modal" data-target="#editModal" unicorn:click="set_form('{{ kid.id }}')">
                                    <i class="tim-icons icon-heart-2"></i>
                                </button>
                            </td>
                        </tr>
                        {% empty %}
                        <tr>
                            <td>
                                No Records
                            </td>
                            <td>
                                
                            </td>
                            <td>
                                
                            </td>
                            <td class="text-center">
                                
                            </td>
                        </tr>
                        {% endfor %}
                    </tbody>
                    </table>
                </div>
                </div>
            </div>
        </div>

    </div>
    
</div>
`

`#the error in console
Internal Server Error: /unicorn/message/memberslist
Traceback (most recent call last):
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "<decorator-gen-25>", line 2, in message
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django_unicorn/decorators.py", line 21, in timed
    result = func(*args, **kwargs)
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django_unicorn/views/__init__.py", line 44, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django/utils/decorators.py", line 130, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django_unicorn/views/__init__.py", line 528, in message
    json_result = _handle_component_request(request, component_request)
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django_unicorn/views/__init__.py", line 374, in _handle_component_request
    return _process_component_request(request, component_request)
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django_unicorn/views/__init__.py", line 151, in _process_component_request
    component_request.data = orjson.loads(component.get_frontend_context_variables())
  File "<decorator-gen-8>", line 2, in get_frontend_context_variables
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django_unicorn/decorators.py", line 21, in timed
    result = func(*args, **kwargs)
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django_unicorn/components/unicorn_view.py", line 476, in get_frontend_context_variables
    encoded_frontend_context_variables = serializer.dumps(
  File "/home/devops/Desktop/cfcic-kids/venv/lib/python3.10/site-packages/django_unicorn/serializer.py", line 301, in dumps
    serialized_data = orjson.dumps(data, default=_json_serializer)
TypeError: Type is not JSON serializable: EditForm
[07/Nov/2022 12:24:37] "POST /unicorn/message/memberslist HTTP/1.1" 500 19192

`
